npm/design-system/src/components/SpecList/SpecList.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import {
+  FileComponentProps,
+  FolderComponentProps,
+  FileExplorer,
+  FileExplorerProps,
+} from '../FileExplorer/FileExplorer'
+import { makeFileHierarchy } from '../FileExplorer/helpers/makeFileHierarchy'
+
+import { InlineIcon } from '@iconify/react'
+import javascriptIcon from '@iconify/icons-vscode-icons/file-type-js-official'
+import typescriptIcon from '@iconify/icons-vscode-icons/file-type-typescript-official'
+import reactJs from '@iconify/icons-vscode-icons/file-type-reactjs'
+import reactTs from '@iconify/icons-vscode-icons/file-type-reactts'
+import folderClosed from '@iconify/icons-vscode-icons/default-folder'
+import folderOpen from '@iconify/icons-vscode-icons/default-folder-opened'
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { fas } from '@fortawesome/free-solid-svg-icons'
+import styles from './SpecList.module.scss'
+
+library.add(fas)
+
+const icons: Record<string, any> = {
+  js: { icon: javascriptIcon },
+  ts: { icon: typescriptIcon },
+  tsx: { icon: reactTs },
+  jsx: { icon: reactJs },
+  folderOpen: { icon: folderOpen },
+  folderClosed: { icon: folderClosed },
+}
+
+const getExt = (path: string) => {
+  const extensionMatches = path.match(/(?:\.([^.]+))?$/)
+
+  return extensionMatches ? extensionMatches[1] : ''
+}
```

Comment:
This should return `undefined` (or `null`, if you want to fight with Typescript a lot) when there's no match, not an empty string